sympy/tensor/array/nditer.py

```diff
@@ -0,0 +1,58 @@
+from sympy.tensor.array.dense_ndim_array import DenseNDimArray
+from sympy.tensor.array.sparse_ndim_array import SparseNDimArray
+from sympy.tensor.array.ndim_array import NDimArray
+from sympy.core.compatibility import Iterable
+
+class nditer(object):
+
+    def __init__(self, iterable):
+        from sympy.matrices.matrices import MatrixBase
+
+        if not isinstance(iterable, (Iterable, MatrixBase)):
+            raise NotImplementedError("Data type not yet supported")
+        if isinstance(iterable, list) and isinstance(iterable[0], NDimArray):
```

Comment:
why this condition?

Reply:
This condition is to handle to case where an array is cast to list by `list()`:
```python
>>> a = list(Array([[0, 1], [2, 3]]))
>>> a.__class__
<class 'list'>
>>> a[0].__class__
<class 'sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray'>
```
And the multdimensional `list` is not considered here, as you explained below. I was thinking the same so I only handled this special case of having a mutipledimensional `list` from `list()`.